code/espurna/sensor.cpp

```diff
@@ -2059,6 +2069,40 @@ void _sensorLoad() {
     }
     #endif
 
+    #if PZEM004TV30_SUPPORT
+    {
+        PZEM004TV30Sensor * sensor = new PZEM004TV30Sensor();
+        _sensors.push_back(sensor);
+
+        // TODO: we need an equivalent to the `pzem.address` command
+        sensor->setAddress(getSetting("pzemv30Addr", PZEM004TV30Sensor::DefaultAddress));
+        sensor->setReadTimeout(getSetting("pzemv30ReadTimeout", PZEM004TV30Sensor::DefaultReadTimeout));
+        sensor->setDebug(getSetting("pzemv30Debug", true));
+
+        uint8_t tx = getSetting("pzemv30TX", PZEM004TV30_SOFTWARE_SERIAL_TX);
+        uint8_t rx = getSetting("pzemv30RX", PZEM004TV30_SOFTWARE_SERIAL_RX);
+        if ((tx != GPIO_NONE) && (rx != GPIO_NONE)) {
+            auto* ptr = new SoftwareSerial(rx, tx);
+            sensor->setDescription("SwSerial");
+            sensor->setStream(ptr); // we don't care about lifetime
+            ptr->begin(PZEM004TV30Sensor::Baudrate);
+        } else {
+            sensor->setDescription("HwSerial");
+            sensor->setStream(&Serial); // note that Serial1 does not receive any data
+            Serial.begin(PZEM004TV30Sensor::Baudrate);
+            // Core does not allow us to begin(baud, cfg, rx, tx) / pins(rx, tx) before begin(baud)
+            // b/c internal UART handler does not exist yet
+            // Also see https://github.com/esp8266/Arduino/issues/2380 as to why there is flush()
+            if (getSetting("pzemv30HwsSwap", 1 == PZEM004TV30_HARDWARE_SERIAL_SWAP)) {
```

Comment:
This is for debugging only, configuration is very confusing right now...
Maybe
- have `pzemv30Soft` 0 or 1, like existing `pzemSoft`, instead of this magic
- make `pzemv30RX` and `TX` checks for 13 and 15, 1 and 3 as valid HardwareSerial pins, don't do anything otherwise.